File: app/src/main/java/test/alexzander/swipetodelete/ContactHolder.java
package test.alexzander.swipetodelete;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

/**
 * Created by AlexZandR on 4/24/17.
 */

public class ContactHolder extends RecyclerView.ViewHolder {

    public boolean isPendingDelete = false;
    public TextView deletedName;
    public TextView name;
    public TextView phone;
    public View progressIndicator;
    public View contactContainer;
    public View undoContainer;
    public View undoData;


    public ContactHolder(final View itemView, final UndoClickListener listener) {
        super(itemView);
        deletedName = (TextView) itemView.findViewById(R.id.user_name_deleted);
        name = (TextView) itemView.findViewById(R.id.user_name);
        phone = (TextView) itemView.findViewById(R.id.user_phone_number);
        contactContainer = itemView.findViewById(R.id.contact_container);
        undoContainer = itemView.findViewById(R.id.undo_container);
        undoData = itemView.findViewById(R.id.undo_data);
        itemView.findViewById(R.id.button_undo).setOnClickListener(
                new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        listener.onUndoClick(getAdapterPosition());
                    }
                });
        progressIndicator = itemView.findViewById(R.id.progress_indicator);
    }

    View getTopContainer() {
        if (isPendingDelete) {
            return undoContainer;
        } else {
            return contactContainer;
        }
    }
}

File: app/src/main/java/test/alexzander/swipetodelete/ContactHolder.kt
package test.alexzander.swipetodelete

import android.support.v7.widget.RecyclerView
import android.view.View
import kotlinx.android.synthetic.main.list_item.view.*


/**
 * Created by AlexZandR on 4/24/17.
 */

class ContactHolder(view: View, listener: UndoClickListener) : RecyclerView.ViewHolder(view) {

    var isPendingDelete = false
    var deletedName = view.user_name_deleted
    var name = view.user_name
    var phone = view.user_phone_number
    var progressIndicator = view.progress_indicator
    var contactContainer = view.contact_container
    var undoContainer = view.undo_container
    var undoData = view.undo_data

    init {
        view.button_undo.setOnClickListener { listener.onUndoClick(adapterPosition) }
    }

    val topContainer: View
        get() =
        if (isPendingDelete) undoContainer
        else contactContainer
}